src/gluonts/dataset/schema/translate.py

```diff
@@ -307,21 +326,32 @@ class Translator:
     """
 
     fields: Dict[str, Op]
+    drop: bool = False
 
     @staticmethod
-    def parse(fields: Optional[dict] = None, **kwargs_fields) -> "Translator":
+    def parse(
+        fields: Optional[dict] = None, drop: bool = False, **kwargs_fields
+    ) -> "Translator":
         fields_ = {}
         if fields is not None:
             fields_.update(fields)
 
         fields_.update(kwargs_fields)
 
-        return Translator(valmap(parse, fields_))
+        return Translator(valmap(parse, fields_), drop)
 
     def __call__(self, item):
-        result = dict(item)
+        if self.drop:
+            keys = item.keys() - self.get_fields()
+            result = {key: item[key] for key in keys}
+        else:
+            result = dict(item)
+
```

Comment:
What is supposed to happen when we call the `Translator` with `drop=True`? I could imagine two things:
- Dropping `item` from `fields` or
- returning the keys `fields` that are not in `item`

It seems to me like `item` would be a subset of `fields` so `item.keys() - self.get_fields()` would always be empty...

Reply:
`item` is the input, so not sure what you mean with dropping item from fields.

The idea is that if `drop` is enabled, only those fields are copied to the result which are not selected by some translation.

This is what it does:

```py
t = Translator.parse(a="b", drop=True)
t(
    {
        "b": 1, # will be removed if `drop` because selected by a
        "c": 42, # will remain always
    }
)
````

Reply:
I see, that makes sense. 👍
Could you add a similar explanation to the docstring, explaining what `drop` does?